src/js/helpers/Repo.js

```diff
@@ -419,7 +420,35 @@ export default class Repo {
    * @return {Promise<void>}
    */
   async save(message) {
-    await this.addAll();
-    await this.commit(message, makeAuthor(this.user));
+    doingSave = true;
+    console.log("Repo.save(" + message + ")");
+    try {
+      // remove deleted files
+      const stagedFiles = await git.listFiles({
+        dir: this.dir
+      });
+
+      for (let i = 0, len = stagedFiles.length; i < len; i++) {
+        const filePath = path.join(this.dir, stagedFiles[i]);
+        const deleted = !fs.existsSync(filePath);
+        if (deleted) { // if deleted from file system, do double check if git needs to remove
+          const status = await this.status(stagedFiles[i]);
```

Comment:
It should be just as fast/slow to go ahead and remove the file. So let's just replace this whole inner if-block with
`await this.remove(stagedFiles[i]);`

Reply:
@PhotoNomad0 I just had another thought. What if we extend our `this.status` with `fs.existsSync`?
So before running `git.status` we'll check if the file exists. If it doesn't we'll assume it's deleted and return the string `*deleted` from `this.status`.
This would be a lot cleaner approach.